src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs

```diff
@@ -49,16 +48,16 @@ static PowerShellProcessInstance()
         /// <param name="useWow64"></param>
         public PowerShellProcessInstance(Version powerShellVersion, PSCredential credential, ScriptBlock initializationScript, bool useWow64)
         {
-            string psWow64Path = s_PSExePath;
+            string psWow64Path = PwshExePath;
 
             if (useWow64)
             {
                 string procArch = Environment.GetEnvironmentVariable("PROCESSOR_ARCHITECTURE");
 
-                if ((!string.IsNullOrEmpty(procArch)) && (procArch.Equals("amd64", StringComparison.OrdinalIgnoreCase) ||
-                    procArch.Equals("ia64", StringComparison.OrdinalIgnoreCase)))
+                if (!string.IsNullOrEmpty(procArch) && (procArch.Equals("amd64", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
@adityapatwardhan does this need to be updated for ARM?

Reply:
Yes, but what is the expected behavior on using `useWow64` when run on arm64?

Reply:
I don't think we should support the `-RunAs32` parameter on `Start-Job` anymore. To start a 32-bit process pwsh, it requires the win-32 pwsh package to be installed, and there is no way to guarantee that, and I don't think we should even bother to search for that installation.

If you look at the current implementation closely, you will find that when `useWow64` is `true`, we are actually using the same default pwsh under `$PSHOME`. So it's already broken. No one reports it because no one is using it.

I think when `-RunAs32` is specified in `Start-Job`, we check if the current process is 64-bit, if so, we throw terminating error. If not, then we continue to run because the underlying pwsh is a 32-bit executable anyways.

Reply:
I agree that the `-RunAs32` is probably not used much anymore since the world has moved to 64-bit and makes even less sense for pwsh

Reply:
We can do that in a separate PR and not block this one

Reply:
Agreed. I will submit a separate PR for that.